gokart/task.py

```diff
@@ -55,14 +55,38 @@ def __init__(self, *args, **kwargs):
         self.task_unique_id = None
         super(TaskOnKart, self).__init__(*args, **kwargs)
         self._rerun_state = self.rerun
+        self.requires = self._add_task_instance(self.requires)
 
     def output(self):
         file_path = self.__module__.replace(".", "/")
         return self.make_target(os.path.join(file_path, f"{type(self).__name__}.pkl"))
 
     def requires(self):
-        tasks = self.make_task_instance_dictionary()
-        return tasks or []  # when tasks is empty dict, then this returns empty list.
+        return {}
```

Comment:
@mski-iksm I think this implementation is a little bit tricky and confusing because of adding requirements from outside of overridden requires().

If you need that feature, you can explicitly write requires() as follows.
```
def OvrerrideTask(TaskOnKart):
   def requires(self):
     super().requires() + dict(other_task=...)
``` 
or
```
def OvrerrideTask(TaskOnKart):
   def requires(self):
     self.make_task_instance_dictionary() + dict(other_task=...)
```

Reply:
@mski-iksm I think this implementation is a little bit tricky and confusing because of adding requirements from outside of overridden requires().

If you need that feature, you can explicitly write requires() as follows.
```python
def OvrerrideTask(TaskOnKart):
   def requires(self):
     super().requires() + dict(other_task=...)
``` 
or
```python
def OvrerrideTask(TaskOnKart):
   def requires(self):
     self.make_task_instance_dictionary() + dict(other_task=...)
```